python/lsst/rubintv/production/plotting/latissNightReportPlots.py

```diff
@@ -0,0 +1,1000 @@
+        ax = plt.gca()
+        xfmt = md.DateFormatter('%m-%d %H:%M:%S')
+        ax.xaxis.set_major_formatter(xfmt)
+        ax.tick_params(which='both', direction='in')
+        ax.minorticks_on()
+        plt.legend()
+        return True
+
+
+class PsfE1Plot(LatissPlot):
+    _PlotName = 'PSF-e1'
+    _PlotGroup = 'Seeing'
+
+    def __init__(self,
+                 dayObs,
+                 locationConfig=None,
+                 uploader=None):
+        super().__init__(dayObs=dayObs,
+                         plotName=self._PlotName,
+                         plotGroup=self._PlotGroup,
+                         locationConfig=locationConfig,
+                         uploader=uploader)
+
+    def plot(self, nightReport, metadata, ccdVisitTable):
+        """Plot the PSF ellipticity e1 values for the current report.
+
+        Parameters
+        ----------
+        nightReport : `lsst.rubintv.production.nightReport.NightReport`
+            The night report for the current night.
+        metadata : `pandas.DataFrame`
+            The front page metadata, as a dataframe.
+        ccdVisitTable : `pandas.DataFrame`
+            The visit summary table for the current day.
+
+        Returns
+        -------
+        success : `bool`
+            Did the plotting succeed, and thus upload should be performed?
+        """
+        # TODO: get a figure you can reuse to avoid matplotlib memory leak
+        plt.figure(constrained_layout=True)
+
+        datesDict = nightReport.getDatesForSeqNums()
+
+        for item in ['PSF e1', 'Filter']:
+            if item not in metadata.columns:
+                msg = f'Cannot create {self._PlotName} plot as required item {item} is not in the table.'
+                self.log.warning(msg)
+                return False
+
+        inds = metadata.index[metadata['PSF e1'] > -100].tolist()  # get the non-nan values
+        rawDates = np.asarray([datesDict[seqNum] for seqNum in inds])
+        psf_e1 = np.array(metadata['PSF e1'][inds])
+        bands = np.array(metadata['Filter'][inds])
+
+        # TODO: generalise this to all bands
+        rband = np.where(bands == 'SDSSr_65mm')
+        gband = np.where(bands == 'SDSSg_65mm')
+        iband = np.where(bands == 'SDSSi_65mm')
+
+        plt.plot(rawDates[gband], psf_e1[gband], '.', color=gcolor, linestyle='-', label='SDSSg')
+        plt.plot(rawDates[rband], psf_e1[rband], '.', color=rcolor, linestyle='-', label='SDSSr')
+        plt.plot(rawDates[iband], psf_e1[iband], '.', color=icolor, linestyle='-', label='SDSSi')
+
+        plt.xlabel('TAI Date')
+        plt.ylabel('PSF Ellipticity e1')
+        plt.xticks(rotation=25, horizontalalignment='right')
+        plt.grid()
+        ax = plt.gca()
+        xfmt = md.DateFormatter('%m-%d %H:%M:%S')
+        ax.xaxis.set_major_formatter(xfmt)
+        ax.tick_params(which='both', direction='in')
+        ax.minorticks_on()
+        plt.legend()
+        return True
+
+
+class PsfE2Plot(LatissPlot):
```

Comment:
I'm not commenting about plotting code duplication for the most part on this PR, because I think fixing that kind duplication is often a deceptively hard problem, and very often the solutions are much worse than the problem.

But needing separate classes for plotting e1 and e2 is a pretty clear indication that something is wrong.  Happy to pair-code if you feel boxed in on this; it's not clear enough to me what led to this to propose a solution.

Reply:
As discussed OOB, this can now be sorted quite easily, and will be addressed as needed on another ticket.